crypto/src/main/java/org/hyperledger/besu/crypto/KeyPairSecurityModule.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.crypto;
+
+import static org.hyperledger.besu.crypto.ECPointUtil.fromBouncyCastleECPoint;
+
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModule;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModuleException;
+import org.hyperledger.besu.plugin.services.securitymodule.data.PublicKey;
+import org.hyperledger.besu.plugin.services.securitymodule.data.Signature;
+
+import java.math.BigInteger;
+import java.security.spec.ECPoint;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * Implementation of SecurityModule wrapping SECP256K1.KeyPair and performing cryptographic
+ * operations via SECP256K1 class
+ */
+public class KeyPairSecurityModule implements SecurityModule {
+  private final SECP256K1.KeyPair keyPair;
+  private final PublicKey publicKey;
+
+  public KeyPairSecurityModule(final SECP256K1.KeyPair keyPair) {
```

Comment:
nit: given there's crypto operations happening here, do we want to move this to a static creator? which passes in a privkey and a PublicKey?

Reply:
we only require SECP256K1.KeyPair. The PublicKey is the interface (with ECPoint) that is meant to be converted from SECP256K1's PublicKey. The other option would be to initialize public key on the first access via `getPublicKey`. I don't feel the need of creator factory method.